src/bucket.ts

```diff
@@ -1789,6 +1798,97 @@ class Bucket extends ServiceObject {
     );
   }
 
+  enableLogging(
+    config: EnableLoggingOptions
+  ): Promise<SetBucketMetadataResponse>;
+  /**
+   * Configuration object for enabling logging.
+   *
+   * @typedef {object} EnableLoggingOptions
+   * @property {string|Bucket} [bucket] The bucket for the log entries. By
+   *     default, the current bucket is used.
+   * @property {string} prefix A unique prefix for log object names.
+   */
+  /**
+   * Enable logging functionality for this bucket. This will make two API
+   * requests, first to grant Cloud Storage WRITE permission to the bucket, then
+   * to set the appropriate configuration on the Bucket's metadata.
+   *
+   * @param {EnableLoggingOptions} config Configuration options.
+   * @param {SetBucketMetadataCallback} [callback] Callback function.
+   * @returns {Promise<SetBucketMetadataResponse>}
+   *
+   * @example
+   * const {Storage} = require('@google-cloud/storage');
+   * const storage = new Storage();
+   * const bucket = storage.bucket('albums');
+   *
+   * const config = {
+   *   prefix: 'log'
+   * };
+   *
+   * bucket.enableLogging(config, function(err, apiResponse) {
+   *   if (!err) {
+   *     // Logging functionality enabled successfully.
+   *   }
+   * });
+   *
+   * @example <caption>Optionally, provide a destination bucket.</caption>
+   * const config = {
+   *   prefix: 'log',
+   *   bucket: 'destination-bucket'
+   * };
+   *
+   * bucket.enableLogging(config, function(err, apiResponse) {});
+   *
+   * @example <caption>If the callback is omitted, we'll return a Promise.
```

Comment:
@callmehiphop Ran into the end of the line. How's this looking?

Reply:
I actually think you have to run over the end in this case. Pretty sure our docs aren't smart enough to know that the closing tag is part of the caption still.